clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -538,14 +538,78 @@ check_duplicate_syscall_with_same_pc()
     return true;
 }
 
+bool
+check_rseq_side_exit_discontinuity()
+{
+#if defined(X86_64) || defined(X86_32) || defined(ARM_64)
+    constexpr addr_t ADDR_ONE = 0xeba4ad4;
+    constexpr addr_t ADDR_TWO = 0xeba4ae4;
+    // Negative test: branches that do not go to their targets due to rseq side
+    // exit.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_marker(1, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+#    if defined(X86_64) || defined(X86_32)
+            // eba4ad4:     74 32        je     11a9ff29
+            gen_branch_encoded(1, ADDR_ONE, { 0x74, 0x32 }),
+            // eba4ad8:     0f 11 40 10         movups %xmm0,0x10(%rax)
+            gen_instr_encoded(ADDR_ONE + 2, { 0x0f, 0x11, 0x40, 0x10 }),
```

Comment:
As Derek suggested offline, we should also add the memrefs for this memory-accessing instruction.

Reply:
Done.